Interop/ModelCard/src/main/java/ModelDetails.java

```diff
@@ -0,0 +1,115 @@
+import com.oracle.labs.mlrg.olcut.provenance.ConfiguredObjectProvenance;
+import org.tribuo.Model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ModelDetails {
+    private static final ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);;
+    private final String schemaVersion;
+    private final String modelType;
+    private final String modelPackage;
+    private final String tribuoVersion;
+    private final String javaVersion;
+    private final Map<String, Object> configuredParams = new HashMap<>();
+
+    public ModelDetails(Model<?> model) {
+        schemaVersion = "1.0";
+        modelType = model.getClass().getSimpleName();
+        modelPackage = model.getClass().getTypeName();
+        tribuoVersion = model.getProvenance().getTribuoVersion();
+        javaVersion = model.getProvenance().getJavaVersion();
+
+        Map<String,?> parameters = model.getProvenance().getTrainerProvenance().getConfiguredParameters();
+        for (String key : parameters.keySet())
+            configuredParams.put(key, parameters.get(key));
+    }
+
+    public ModelDetails(JsonNode modelDetailsJson) throws JsonProcessingException {
+        schemaVersion = modelDetailsJson.get("schema-version").textValue();
+        modelType = modelDetailsJson.get("model-type").textValue();
+        modelPackage = modelDetailsJson.get("model-package").textValue();
+        tribuoVersion = modelDetailsJson.get("tribuo-version").textValue();
+        javaVersion = modelDetailsJson.get("java-version").textValue();
+        Map<?,?> params = mapper.readValue(modelDetailsJson.get("configured-parameters").toString(), Map.class);
+        for (var key : params.keySet())
+            configuredParams.put(key.toString(), params.get(key));
+    }
+
+    public String getSchemaVersion() {
+        return schemaVersion;
+    }
+
+    public String getModelType() {
+        return modelType;
+    }
+
+    public String getModelPackage() {
+        return modelPackage;
+    }
+
+    public String getTribuoVersion() {
+        return tribuoVersion;
+    }
+
+    public String getJavaVersion() {
+        return javaVersion;
+    }
+
+    public Map<String, Object> getConfiguredParams() {
+        return configuredParams;
+    }
+
+    public ObjectNode toJson() {
+        ObjectNode modelDetailsObject = mapper.createObjectNode();
+        modelDetailsObject.put("schema-version", schemaVersion);
+        modelDetailsObject.put("model-type", modelType);
+        modelDetailsObject.put("model-package", modelPackage);
+        modelDetailsObject.put("tribuo-version", tribuoVersion);
+        modelDetailsObject.put("java-version", javaVersion);
+        ObjectNode paramsArr = paramsToJson(null, configuredParams);
+        modelDetailsObject.set("configured-parameters", paramsArr);
+        return modelDetailsObject;
+    }
+
+    private ObjectNode paramsToJson(String name, Map<?,?> params) {
```

Comment:
Got it - I've now introduced a new field `JsonNode configuredParams` so that this process can instead happen in the constructor. Because of this, I also no longer saw the need to have a `Map configuredParams` (since they would both essentially contain the same info). So, I've removed `Map configuredParams` for now, but if there is a reason for keeping it, I can always add it back.